src/components/common/EthHashInfo/index.tsx

```diff
@@ -83,8 +83,8 @@ const PrefixedEthHashInfo = ({
   const settings = useAppSelector(selectSettings)
   const chain = useCurrentChain()
   const addressBook = useAddressBook()
-  // prefer address book name
-  const name = showName ? addressBook[props.address] || props.name : undefined
+
+  const name = showName ? props.name || addressBook[props.address] : undefined
```

Comment:
We use this component to render owners on the final load Safe step. Without this change it renders the address book entry instead of the desired name.

I think this order makes more sense anyway: if we are passing `props.name` then we explicitly want to use that.

Reply:
The only other place we use the `name` props (outside of load/create Safe) is in the `<TxSigners>` component when displaying the `executor.name`.

Reply:
Alternatively, we could add the address book entry on the previous step. Wdyt @usame-algan? There are a few places that this component receives `name`. All of which are known `name`s that come from backend.